vars/prepareDefaultValues.groovy

```diff
@@ -1,9 +1,24 @@
 import com.sap.piper.DefaultValueCache
+import com.sap.piper.MapUtils
 
 def call(Map parameters = [:]) {
     handlePipelineStepErrors (stepName: 'prepareDefaultValues', stepParameters: parameters) {
-        if(!DefaultValueCache.getInstance()) {
-            Map defaultValues  = readYaml text: libraryResource('default_pipeline_environment.yml')
+        if(!DefaultValueCache.getInstance() || parameters.customDefaults) {
+            def defaultValues = [:]
+            def configurationFiles = ['default_pipeline_environment.yml']
+            def customDefaults = parameters.customDefaults
```

Comment:
Just from the naming, I thought customDefaults contains the values, but it it just the names of the configuration files. 
What about customConfigurationFiles?